pkg/operators/v1alpha1/subscription_types.go

```diff
@@ -118,8 +118,18 @@ const (
 	// SubscriptionBundleUnpackFailed indicates that the unpack job failed
 	SubscriptionBundleUnpackFailed SubscriptionConditionType = "BundleUnpackFailed"
 
-	// SubscriptionOperatorDeprecated indicates that the Operator currently installed with this Subscription has been deprecated.
-	SubscriptionOperatorDeprecated SubscriptionConditionType = "OperatorDeprecated"
+	// SubscriptionDeprecated is a roll-up condition which indicates that the Operator currently installed with this Subscription
+	//has been deprecated. It will be present when any of the three deprecation types (Package, Channel, Bundle) are present.
+	SubscriptionDeprecated SubscriptionConditionType = "Deprecated"
```

Comment:
I’m not seeing the point of this condition. There are three distinct conditions that reflect which resource is deprecated. I don’t see the need of a blanket condition unless there is a usage for it that is not described in the comment or commit. This is why we should start to do proof PR to show how these fields are being used as a part of reviewing process.

Reply:
Or is this supposed to replace OperatorDeprecated?

Reply:
Yes, it's replacing `OperatorDeprecated` as the roll-up condition name. Having the roll-up allows us and any users to determine deprecated status without having to check for all three different sup-types.